code/modules/surgery/organs/internal/heart/heart_anomalock.dm

```diff
@@ -0,0 +1,200 @@
+	COOLDOWN_START(src, survival_cooldown, survival_cooldown_time)
+	addtimer(CALLBACK(src, PROC_REF(notify_cooldown), organ_owner), COOLDOWN_TIMELEFT(src, survival_cooldown))
+
+///Alerts our owner that the organ is ready to do its thing again
+/obj/item/organ/internal/heart/cybernetic/anomalock/proc/notify_cooldown(mob/living/carbon/organ_owner)
+	balloon_alert(organ_owner, "your heart strenghtens")
+	playsound(organ_owner, 'sound/items/eshield_recharge.ogg', 40)
+
+///Returns the mob we are implanted in so that the electricity effect doesn't runtime
+/obj/item/organ/internal/heart/cybernetic/anomalock/proc/get_held_mob()
+	return owner
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/attackby(obj/item/item, mob/living/user, params)
+	if(!istype(item.type, required_anomaly))
+		return ..()
+	if(core)
+		balloon_alert(user, "core already in!")
+		return
+	if(!user.transferItemToLoc(item, src))
+		return
+	core = item
+	balloon_alert(user, "core installed")
+	playsound(src, 'sound/machines/click.ogg', 30, TRUE)
+	add_organ_trait(TRAIT_SHOCKIMMUNE)
+	update_icon_state()
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!core)
+		balloon_alert(user, "no core!")
+		return
+	if(!core_removable)
+		balloon_alert(user, "can't remove core!")
+		return
+	balloon_alert(user, "removing core...")
+	if(!do_after(user, 3 SECONDS, target = src))
+		balloon_alert(user, "interrupted!")
+		return
+	balloon_alert(user, "core removed")
+	core.forceMove(drop_location())
+	if(Adjacent(user) && !issilicon(user))
+		user.put_in_hands(core)
+	core = null
+	remove_organ_trait(TRAIT_SHOCKIMMUNE)
+	update_icon_state()
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/update_icon_state()
+	. = ..()
+	icon_state = initial(icon_state) + (core ? "-core" : "")
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/prebuilt/Initialize(mapload)
+	. = ..()
+	core = new /obj/item/assembly/signaler/anomaly/flux(src)
+	update_icon_state()
+
+/datum/status_effect/voltaic_overdrive
+	id = "voltaic_overdrive"
+	duration = 30 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/anomalock_active
+	show_duration = TRUE
+
+/datum/status_effect/voltaic_overdrive/tick(seconds_between_ticks)
+	. = ..()
+
+	if(owner.health <= owner.crit_threshold)
+		owner.heal_overall_damage(5, 5)
+		owner.adjustOxyLoss(-5)
+		owner.adjustToxLoss(-5)
+
+/datum/status_effect/voltaic_overdrive/on_apply()
+	. = ..()
+	owner.add_movespeed_mod_immunities(type, /datum/movespeed_modifier/damage_slowdown)
+	REMOVE_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+	owner.reagents.add_reagent(/datum/reagent/medicine/coagulant, 5)
+	owner.add_filter("emp_shield", 2, outline_filter(1, "#639BFF"))
+	to_chat(owner, span_revendanger("You feel a burst of energy! It's do or die!"))
+	if(iscarbon(owner))
+		var/mob/living/carbon/carbon_owner = owner
+		carbon_owner.gain_trauma(/datum/brain_trauma/special/tenacity)
```

Comment:
Really stupid scenario I am creating here but probably give this `TRAUMA_RESILIENCE_ABSOLUTE` as the second argument (and also when removing it), in order to prevent this from being removed by Changeling Panacea or implanted Regenerative Cores.